File: pset_1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

void greedy_algorithm(int c);

int main(void)
{
    int c; // Change owed
    do
    {
        c = get_int("Change owed: ");
    }
    while (c < 0);
    greedy_algorithm(c);
}

void greedy_algorithm(int c)
{
    int times = 0;

    while (c > 0)
    {
        if (c >= 25)
        {
            c -= 25;
            times++;
        }
        else if (c >= 10)
        {
            c -= 10;
            times++;
        }
        else if (c >= 5)
        {
            c -= 5;
            times++;
        }
        else
        {
            c -= 1;
            times++;
        }
    }

    printf("%i\n", times);
}

File: pset_6/sentimental-cash/cash.py
def greedy_algorithm(c):
    c = round(c * 100)  # Convert to int for accuracy
    times = 0

    while c > 0:
        if c >= 25:  # Quarters
            c -= 25
        elif c >= 10:  # Dimes
            c -= 10
        elif c >= 5:  # Nickels
            c -= 5
        else:  # Pennies
            c -= 1
        times += 1

    print(times)


def main():
    # Demand positive float input from the user
    while True:
        try:
            c = float(input("Change owed: "))
        except ValueError:
            continue
        if c >= 0.0:
            break

    greedy_algorithm(c)


if __name__ == "__main__":
    main()
